File: Day 19 : Interfaces/Solution.java
import java.util.*;

interface AdvancedArithmetic
{
   int divisorSum(int n);
}

class Calculator implements AdvancedArithmetic
{
    public int divisorSum(int n)
    {
        int sum1 = 0;
        for(int i = 1; i <= n; i++)
        {
            if(n%i==0)
            {
                sum1 += i;
            }
        }
        return sum1;
    }
}

class Solution 
{
    public static void main(String[] args) 
    {
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        scan.close();
        
      	AdvancedArithmetic myCalculator = new Calculator(); 
        int sum = myCalculator.divisorSum(n);
        System.out.println("I implemented: " + myCalculator.getClass().getInterfaces()[0].getName() );
        System.out.println(sum);
    }
}

File: Day 19 : Interfaces/Solution.ts
interface AdvancedArithmetic {
    divisorSum(n: number): number
}

class Calculator implements AdvancedArithmetic {
    static interfaceName = "AdvancedArithmetic";

    divisorSum(n: number): number {
        let sum: number = 0;
        for (let i: number = 1; i <= n; i++) {
            if (n % i === 0) sum += i;
        }
        return sum;
    }
}

function main() {
    const n: number = +readLine();
    const myCalculator: Calculator = new Calculator();
    const sum: number = myCalculator.divisorSum(n);
    console.log("I implemented: " + (myCalculator.constructor as any).interfaceName);
    console.log(sum);
}